scripts/hca-utils/hca_utils/utils.py

```diff
@@ -0,0 +1,332 @@
+import csv
+from functools import lru_cache
+import os
+from requests.exceptions import HTTPError
+from typing import Set
+import urllib.parse
+
+import google.auth
+from google.auth.transport.requests import AuthorizedSession
+from google.cloud import bigquery, storage
+from requests_cache.core import CachedSession
+
+
+class HcaUtils:
+    def __init__(self, environment: str, project: str, dataset: str):
+        self.environment = environment
+
+        if environment == "dev":
+            self.project = "broad-jade-dev-data"
+        else:
+            self.project = project
+
+        self.dataset = dataset
+
+        self.filename_template = f"sd-{project}-{dataset}-{{table}}.csv"
+
+        self.bigquery_client = bigquery.Client(project=self.project)
+
+        bucket_projects = {"prod": "mystical-slate-284720",
+                           "dev": "broad-dsp-monster-hca-dev"}
+        self.bucket_project = bucket_projects[environment]
+        self.bucket = f"broad-dsp-monster-hca-{environment}-staging-storage"
+
+        jade_urls = {"prod": "https://jade-terra.datarepo-prod.broadinstitute.org",
+                     "dev": "https://jade.datarepo-dev.broadinstitute.org"}
+        self.base_url = jade_urls[environment]
+
+        # this is always prod for bigquery, storage, etc.
+        gcp_creds, _ = google.auth.load_credentials_from_file("path/to/prod/credentials.json")
```

Comment:
I am not proud of this

Reply:
if you import `from google.oauth2.service_account import ServiceAccount` you can do this instead:
```suggestion
        self.gcp_creds = service_account.Credentials.from_service_account_file("path/to/prod/credentials.json")
```